src/test-helper.ts

```diff
@@ -0,0 +1,67 @@
+import {mockTimeSource} from './mock-time-source';
+
+function makeTestHelper (options: Object = {}) {
+  return function withTime (test, testOptions: Object = {}) {
+    return function runTestWithTime (done) {
+      const Time = mockTimeSource({...options, ...testOptions});
+
+      test(Time);
+
+      // It's hard to differentiate between tape and ava
+      const usingAva = !!done && typeof done.regex === 'function';
```

Comment:
Yeah, good call. I'll still need some special logic, ava doesn't "just work" in promise mode unfortunately (complains about no assertions being called).

I'm tempted to recommend ava users just use callback mode because it's simpler to support.